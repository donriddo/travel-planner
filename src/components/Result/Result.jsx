import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

class Result extends React.Component {

    componentDidMount() {

    }

    render() {
        const { distance, from, to, date, numOfPassengers } = this.props.search;
        return (
            <div>
                { distance && <p>Distance: {parseFloat(distance)/1000.0} km</p> }
                { from && <p>from: {from}</p> }
                { to && <p>to: {to}</p> }
                { date && <p>date: {date}</p> }
                { numOfPassengers && <p>numOfPassengers: {numOfPassengers}</p> }
            </div>
        );
    }
}

function mapStateToProps(state) {
    const { search } = state;
    return {
        search
    };
}

export default connect(mapStateToProps)(Result);