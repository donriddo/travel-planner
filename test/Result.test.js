import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import Result from '../src/components/Result/Result';
import { setupGoogleMock } from './setup';

const mockStore = configureMockStore();
const store = mockStore({
    search: { start: '', end: '', passengers: '', date: '' }
});

beforeAll(() => {
  setupGoogleMock();
});

test('Result test', () => {
  const component = renderer.create(
    <Provider store={store}>
        <Result location={{}} />
    </Provider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});