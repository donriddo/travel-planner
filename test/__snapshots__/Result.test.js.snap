// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result test 1`] = `
<div
  className="result"
>
  <p>
    Distance: 
    0
     km
  </p>
  
  
  
  <p>
    Number of Passengers: 
    0
  </p>
</div>
`;
